#1777: FML Parser in StructureMapUtilities and id transform

oliveregger opened the pull request:
That the test run through the PR https://github.com/FHIR/fhir-test-cases/pull/195 needs to be integrated.

See description for PR in https://github.com/hapifhir/org.hl7.fhir.core/issues/1777

grahamegrieve reviewed (commented):
what's with this bit:
````java
  if (!(context.getVersion().equals("5.0.0"))) {
    throw new FHIRException("FHIR version needs to be 5.0.0");
  }
````
     
     ??

grahamegrieve reviewed (commented):
Also, ObjectConverter has a method to do the ElementModel -> Resource conversion

oliveregger commented:
> what's with this bit:
> 
> ```java
>   if (!(context.getVersion().equals("5.0.0"))) {
>     throw new FHIRException("FHIR version needs to be 5.0.0");
>   }
> ```
> 
> ??

I added this, because we use the R5 functionality in an R4 instance. StructureMapUtilities does not work with an R4 context and after noticing the problem, I added this safeguard. but no problem in removing it.

oliveregger commented:
the narrative generation has also changed with FML Parser, that the test run through, I updated the testcase here:
https://github.com/FHIR/fhir-test-cases/pull/211

grahamegrieve reviewed (approved):
ok
